packages/design-system/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,187 @@
+import Ellipses from './Ellipses'
+import Page from './Page'
+import React, {useState} from 'react'
+
+const overflow = 1;
+const maxVisiblePages = 7;
+
+function paginationBuilder(currentPage, totalPages) {
+  const range = [];
+
+  let start = currentPage - overflow;
+  let end = currentPage + overflow;
+
+  /**
+   * If `currentPage` = 1, only add first three pages to `range[]`
+   */
+  if (start < 1) {
+    start = 1;
+    end = start + overflow * 2; // 3
+  }
+
+  /**
+   * If `end` > `totalPages`, only add last three pages to `range[]`
+   */
+  if (end > totalPages) {
+    start = totalPages - overflow * 2; 
+    end = totalPages;
+  }
+
+  /**
+   * If `totalPages` is 5 or fewer, all pages added to `range[]`
+   */
+  if (totalPages <= maxVisiblePages) {
+    start = 1;
+    end = totalPages;
+  }
+
+  for (let i = start; i <= end; i++) {
+    range.push(i);
+  }
+  
+  return range;
+}
+
+export default function Pagination({compact, customUrl, page, totalPages}) {
+  const [currentPage, setCurrentPage] = useState(page)
+  const pageRange = paginationBuilder(currentPage, totalPages)
+  const pages = [];
+
+  if (pageRange[0] >= 2) {
+    /**
+     * If `pageRange` begins with a page of 2 or greater,
+     * begin Pagination with Page 1 
+     */
+    pages.push(
+      <Page
+        customUrl={customUrl}
+        key="page-1"
+        index={1}
+        currentPage={currentPage}
+        onPageChange={() => setCurrentPage(1)}
+      />
+    )
+
+    /**
+     * If `pageRange` doesn't equal 2, second Pagination element is Ellipses,
+     * otherwise page count continues.
+     */
+    if (pageRange[0] !== 2) {
+      pages.push(<Ellipses key="ellipses-1" />)
+    }
+  }
+
+  /**
+   * Renders all Page components in range (3 pages) to Pagination component.
+   */ 
+  pageRange.map(p => {
+    pages.push(
+      <Page
+        customUrl={customUrl}
+        key={`page-${p}`}
+        index={p}
+        currentPage={currentPage}
+        onPageChange={() => setCurrentPage(p)}
+      />
+    )
+  });
+
+  /**
+   * Defines if/when the Ellipses component appears
+   * at the end of the Pagination component - 
+   * as long as there are fewer than 7 pages.
+   */
+  if (currentPage <= totalPages - overflow - 1 && totalPages > maxVisiblePages) {
+    if (currentPage < totalPages - overflow - 1) {
+      pages.push(<Ellipses key="ellipses-2" />)
+    }
+
+    pages.push(
+      <Page
+        customUrl={customUrl}
+        key={`page-${totalPages}`}
+        index={totalPages}
+        currentPage={currentPage}
+        onPageChange={() => setCurrentPage(totalPages)}
+      />
+    )
+  }
+
+  return (
+    <nav 
+      className="ds-c-pagination" 
+      role="navigation" 
+      aria-label="Pagination Navigation"
+    >
+      { currentPage !== 1 && (
+        <a 
```

Comment:
I agree - I'll look into how to address this stylistically

Reply:
I'm thinking we could use - `visibility: hidden; to hide the previous/next buttons or other elements. From what I understand that would also be effective for screen readers as well

Reply:
I agree this could be solved with style updates, but I recommend holding off on implementing any fix just yet - there's some a11y stuff I need to chat with DK about once this wraps that might change the markup.